pkg/controllers/monitoring/monitoring-stack/controller.go

```diff
@@ -143,17 +151,50 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 			return ctrl.Result{RequeueAfter: 2 * time.Second}, nil
 		}
 		if err != nil {
-			return ctrl.Result{}, err
+			return r.updateConditions(ctx, req, ms)
 		}
 	}
-	return ctrl.Result{}, nil
+	return r.updateConditions(ctx, req, ms)
+}
+
+func (r *reconciler) updateConditions(ctx context.Context, req ctrl.Request, ms *stack.MonitoringStack) (ctrl.Result, error) {
+	prom, err := r.mcClient.MonitoringV1().Prometheuses(req.Namespace).Get(ctx, ms.Name, metav1.GetOptions{})
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	newStatus := stack.MonitoringStackStatus{}
+	for i := range prom.Status.Conditions {
+		pc := prom.Status.Conditions[i]
+		c := newPrometheusCondition(pc, ms.Generation)
+		newStatus.Conditions = append(newStatus.Conditions, c)
+	}
+	ms.Status = newStatus
+	err = r.k8sClient.Status().Update(ctx, ms)
+	if errors.IsConflict(err) {
+		return ctrl.Result{RequeueAfter: 2 * time.Second}, nil
+	}
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	return ctrl.Result{RequeueAfter: 1 * time.Minute}, nil
+
+}
+
+func newPrometheusCondition(pc monv1.PrometheusCondition, generation int64) metav1.Condition {
+	return metav1.Condition{
+		Type:               string(pc.Type),
+		Message:            pc.Message,
+		Reason:             pc.Reason,
+		LastTransitionTime: pc.LastTransitionTime,
+		Status:             metav1.ConditionStatus(pc.Status),
```

Comment:
This can currently fails because the [`PrometheusConditionStatus`](https://github.com/prometheus-operator/prometheus-operator/blob/main/pkg/apis/monitoring/v1/types.go#L632) doesn't conform to [`Kubernetes ConditionStatus`](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#ConditionStatus) (there's no degraded value).
So the question is what to do with that:
- fix/update in the Prometheus operator? 
- use the `PrometheusCondition` type here in OO? 
- use the current approach but introduce some tweak/hack to make compliant

Reply:
I would vote for the 2nd option but let OBO define its own ConditionStatus with a mapping from PrometheusCondition to ObservabilityCondition. WYDT?

Reply:
> So if I understand it correctly the suggestion is basically to introduce a custom condition type for OBO (something like ObservabilityCondtiion or MonitoringStackCondition) right?

yes it would be my suggestion.